File: V1-extensions/ChatMessageExtension/src/main/java/io/github/toberocat/improvedfactions/listeners/PlayerListener.java
package io.github.toberocat.improvedfactions.listeners;


import io.github.toberocat.improvedfactions.ChatMessageExtension;
import io.github.toberocat.improvedfactions.types.ChatType;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.AsyncPlayerChatEvent;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

import static io.github.toberocat.improvedfactions.modules.ChatModule.send;
import static io.github.toberocat.improvedfactions.modules.ChatModule.sendAlly;

public class PlayerListener implements Listener {

    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
    public void chat(AsyncPlayerChatEvent event) {
        ChatType type = ChatMessageExtension.PLAYER_CHAT_TYPE.get(event.getPlayer().getUniqueId());
        if (type == ChatType.FACTION) {
            send(event.getPlayer(), event.getMessage());
            event.setCancelled(true);
        } else if (type == ChatType.ALLIES) {
            sendAlly(event.getPlayer(), event.getMessage());
            event.setCancelled(true);
        }
    }

    @EventHandler
    public void PlayerJoin(PlayerJoinEvent event) {
        ChatMessageExtension.PLAYER_CHAT_TYPE.put(event.getPlayer().getUniqueId(), ChatType.PUBLIC);
    }

    @EventHandler
    public void PlayerLeave(PlayerQuitEvent event) {
        ChatMessageExtension.PLAYER_CHAT_TYPE.remove(event.getPlayer().getUniqueId(), ChatType.PUBLIC);
    }
}

File: improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/modules/tutorial/listener/PlayerListener.kt
package io.github.toberocat.improvedfactions.modules.tutorial.listener

import io.github.toberocat.improvedfactions.modules.tutorial.manager.TutorialManager
import org.bukkit.event.Cancellable
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerEvent
import org.bukkit.event.player.PlayerMoveEvent

class PlayerListener : Listener {

    @EventHandler
    fun onPlayerMove(event: PlayerMoveEvent) {
        val player = event.player
        if (!TutorialManager.isPlayerMovementLocked(player)) return

        event.isCancelled = true
    }
}
